frontend/src/general/__tests__/DeleteDeviceButton.test.jsx

```diff
@@ -91,65 +94,67 @@ describe("DeleteButton", () => {
-        it("Should call axios.delete, close the dialog and refresh the page when clicked", async () => {
-          // Pretend we are on the device list page
+      describe("If on devices table page", () => {
+        it("Should call axios.delete, close the dialog and refresh the page", async () => {
           mockHistory.location.pathname = "/Devices";
 
-          // click the delete button to set open to true
           wrapper.find("#deleteBtn").simulate("click");
           expect(setOpen).toHaveBeenCalledWith(true);
 
-          // mock axios before clicking confirm
-          const axiosPromise = Promise.resolve();
-          axios.delete.mockImplementationOnce(() => axiosPromise);
+          axios.delete.mockResolvedValueOnce();
 
-          // click confirm
           wrapper.find("#confirmDeleteBtn").simulate("click");
 
-          // check that axios.delete was called
           expect(axios.delete).toHaveBeenCalled();
 
-          // Wait for axios promise to finish
-          await flushPromises();
+          expect(setOpen).toHaveBeenCalledWith(false);
 
-          // check that refresh has been called
-          expect(mockHistoryGo).toHaveBeenCalledWith(0);
+          await flushPromises();
 
-          // open should be false
-          expect(setOpen).toHaveBeenCalledWith(false);
+          expect(mockHistoryGo).toHaveBeenCalledTimes(1);
         });
       });
-
-      describe("on DeviceDetailsPage", () => {
-        it("Should call axios.delete, close the dialog and redirect to device list page when clicked", async () => {
-          // Pretend we are on the device details page
+      describe("If on device details page", () => {
+        it("Should call axios.delete, close the dialog and redirect to /Devices", async () => {
           mockHistory.location.pathname = "Devices/Details/sample_sender";
 
-          // click the delete button to set open to true
           wrapper.find("#deleteBtn").simulate("click");
           expect(setOpen).toHaveBeenCalledWith(true);
 
-          // mock axios before clicking confirm
-          const axiosPromise = Promise.resolve();
-          axios.delete.mockImplementationOnce(() => axiosPromise);
+          axios.delete.mockResolvedValueOnce();
 
-          // click confirm
           wrapper.find("#confirmDeleteBtn").simulate("click");
 
-          // check that axios.delete was called
           expect(axios.delete).toHaveBeenCalled();
 
-          // Wait for axios promise to finish
+          expect(setOpen).toHaveBeenCalledWith(false);
+
           await flushPromises();
 
-          // check that redirect has been called
           expect(mockHistoryPush).toHaveBeenCalledWith("/Devices");
-
-          // open should be false
-          expect(setOpen).toHaveBeenCalledWith(false);
         });
       });
+      it("If the axios.delete is rejected, it should close the dialog and display an error snackbar", async () => {
+        wrapper.find("#deleteBtn").simulate("click");
```

Comment:
Pretty sure we don't use `.simulate("click")` anymore, but I forget what the alternative is when it's a functional component and we can't directly call the function. (I might be mistaken tho)

Reply:
I mentioned in the PR notes that I'm not going to redo the test to fit with our current testing methodology since it's not in the scope of the PR. On top of the fact that we might remove this button entirely very soon to replace the action menu in the device table with the device details icon button.